### PR TITLE
debian: Add initial Debian packaging

### PR DESCRIPTION
**Pull request by pwithnall**
This commit will form the unparented `debian-master` branch when merged, but for the purposes of review I’m setting it to merge into master, because there’s no way to open a pull request to merge into an empty branch.

---

This is for version 2.4b1 of PyNDN2. We can only support python2, not
python3, because some of the dependencies (python-protobuf) don’t
support python3.

Signed-off-by: Philip Withnall <withnall@endlessm.com>
https://phabricator.endlessm.com/T14714

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #! /usr/bin/make -f
 
 #export DH_VERBOSE = 1
-export PYBUILD_NAME = pyndn
```
Comment (pwithnall) [inline debian/rules]: Are you sure it’s safe to remove this? All the [documentation](https://wiki.debian.org/Python/Pybuild#line-33) seems to say it’s required; and looking at `/usr/bin/pybuild` the code seems to set the package name to `PYBUILD_NAME_not_set` if the variable is not set.

File: debian/changelog [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pyndn (2.4b1-0.1) UNRELEASED; urgency=medium
+pyndn (2.4~beta1-1) UNRELEASED; urgency=medium
```
Comment (pwithnall) [inline debian/changelog:1]: That means our package version will not match the upstream version number (`2.4b1`). Do we want that?

File: debian/control [hunk 3]
```diff
@@ -12,18 +12,19 @@ Build-Depends:
  python-trollius,
 X-Python-Version: >= 2.7
 Standards-Version: 3.9.7
-Testsuite: autopkgtest-pkg-python
-Homepage: http://github.com/named-data/PyNDN2
-Vcs-Git: git://github.com/endlessm/PyNDN2.git
-Vcs-Browser: http://github.com/endlessm/PyNDN2
+Testsuite: autopkgtest
```
Comment (pwithnall) [inline debian/control]: This change is incorrect, but you’ve correctly reverted it in the next commit.

**Comment by xaiki** [comment]
so 2.4b1 is incorrect because it's `> 2.4` and we don't want that, `2.4~beta1` will be properly parsed by debian tools and is clearer.

as for PYBUILD_NAME, i had a look at other packages in the archive that were not using it IIRC. and i did try building and it is building it correctly...

having it does not harm this package i guess (IIRC it's what did harm the endless-ndn/edd packages)

**Comment by pwithnall** [comment]
> so 2.4b1 is incorrect because it's > 2.4 and we don't want that, 2.4~beta1 will be properly parsed by debian tools and is clearer.

OK, I’ll change it. Thanks for the explanation.

> as for PYBUILD_NAME, i had a look at other packages in the archive that were not using it IIRC. and i did try building and it is building it correctly...
>
> having it does not harm this package i guess (IIRC it's what did harm the endless-ndn/edd packages)

I’d prefer to keep it if it’s not doing any harm. I can’t immediately see how the `pybuild` code can work without it.

---

Other lintian warnings:

```
W: pyndn source: unknown-testsuite autopkgtest-pkg-python
N:
N:    The dsc file sets Testsuite to a value other than autopkgtest, the only
N:    one allowed. This field is most probably copied by dpkg-source from
N:    Testsuite in debian/control.
N:    
N:    Refer to
N:    http://anonscm.debian.org/gitweb/?p=autopkgtest/autopkgtest.git;a=blob_plain;f=doc/README.package-tests.rst;hb=HEAD
N:    for details.
N:    
N:    Severity: normal, Certainty: certain
N:    
N:    Check: testsuite, Type: source
N:
P: pyndn source: debian-watch-may-check-gpg-signature
N:
N:    This watch file does not include a means to verify the upstream tarball
N:    using cryptographic signature.
N:    
N:    If upstream distributions provide such signatures, please use the
N:    pgpsigurlmangle options in this watch file's opts= to generate the URL
N:    of an upstream GPG signature. This signature is automatically downloaded
N:    and verified against a keyring stored in
N:    debian/upstream/signing-key.asc.
N:    
N:    Of course, not all upstreams provide such signatures, but you could
N:    request them as a way of verifying that no third party has modified the
N:    code against their wishes after the release. Projects such as
N:    phpmyadmin, unrealircd, and proftpd have suffered from this kind of
N:    attack.
N:    
N:    Refer to the uscan(1) manual page for details.
N:    
N:    Severity: pedantic, Certainty: certain
N:    
N:    Check: watch-file, Type: source
N:
W: python-pyndn: new-package-should-close-itp-bug
N:
N:    This package appears to be the first packaging of a new upstream
N:    software package (there is only one changelog entry and the Debian
N:    revision is 1), but it does not close any bugs. The initial upload of a
N:    new package should close the corresponding ITP bug for that package.
N:    
N:    This warning can be ignored if the package is not intended for Debian or
N:    if it is a split of an existing Debian package.
N:    
N:    Refer to Debian Developer's Reference section 5.1 (New packages) for
N:    details.
N:    
N:    Severity: normal, Certainty: certain
N:    
N:    Check: changelog-file, Type: binary
N: 
```

The first two should be added to a lintian ignore file (false positives). The third is valid for upstream, even though we’re not upstreaming this right now.

**Comment by xaiki** [comment]
> >     as for PYBUILD_NAME, i had a look at other packages in the archive that were not using it IIRC. and i did try building and it is building it correctly...
> > having it does not harm this package i guess (IIRC it's what did harm the endless-ndn/edd packages)

> I’d prefer to keep it if it’s not doing any harm. I can’t immediately see how the pybuild code can work without it.

sure, feel free to add it back

**Comment by pwithnall** [comment]
Updated version pushed. Ack/Nack?

**Comment by pwithnall** [comment]
I just rejigged this because I’d incorrectly set the merge destination to `master` rather than `debian-master`.

**Pull request merged**